WcaOnRails/spec/controllers/competitions_controller_spec.rb

```diff
@@ -221,6 +221,23 @@
         sign_in organizer
       end
 
+      it 'cannot pass a non-delegate as delegate' do
+        delegate_ids_old = competition.delegate_ids
+        fake_delegate = FactoryGirl.create(:user)
+        post :update, id: competition, competition: { delegate_ids: fake_delegate.id }
+        invalid_competition = assigns(:competition)
+        expect(invalid_competition.errors.messages[:delegate_ids]).to eq ["You must pass only delegates here"]
+        competition.reload
+        expect(competition.delegate_ids).to eq delegate_ids_old
+      end
+
+      it 'can change the delegate' do
+        new_delegate = FactoryGirl.create(:delegate)
+        post :update, id: competition, competition: { delegate_ids: new_delegate.id }
+        competition.reload
+        expect(competition.delegate_ids).to eq new_delegate.id.to_s
```

Comment:
Actually it doesn't.
expected: #<User id: 5892, email: "stanley@mayert.com", encrypted_passwor
d: "$2a$04$0HN9ywcsGyojngJetOvnqeMzzEP5C8Wm2SHJ...

got: #<ActiveRecord::Associations::CollectionProxy [#<User id: 5892,
 email: "stanley@mayert.com", encrypted_password

I'll keep it the other way.

On Mon, Jan 4, 2016 at 11:51 PM, Jeremy Fleischman <notifications@github.com
> wrote:

> In WcaOnRails/spec/controllers/competitions_controller_spec.rb
> <https://github.com/cubing/worldcubeassociation.org/pull/298#discussion_r48805382>
> :
>
> > @@ -221,6 +221,23 @@
> >          sign_in organizer
> >        end
> >
> > +      it 'cannot pass a non-delegate as delegate' do
> > +        delegate_ids_old = competition.delegate_ids
> > +        fake_delegate = FactoryGirl.create(:user)
> > +        post :update, id: competition, competition: { delegate_ids: fake_delegate.id }
> > +        invalid_competition = assigns(:competition)
> > +        expect(invalid_competition.errors.messages[:delegate_ids]).to eq ["You must pass only delegates here"]
> > +        competition.reload
> > +        expect(competition.delegate_ids).to eq delegate_ids_old
> > +      end
> > +
> > +      it 'can change the delegate' do
> > +        new_delegate = FactoryGirl.create(:delegate)
> > +        post :update, id: competition, competition: { delegate_ids: new_delegate.id }
> > +        competition.reload
> > +        expect(competition.delegate_ids).to eq new_delegate.id.to_s
>
> What about this instead?
>
> expect(competition.delegates).to eq [new_delegate]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cubing/worldcubeassociation.org/pull/298/files#r48805382>
> .
>